lib/rules/semi-spacing.js

```diff
@@ -117,33 +117,55 @@ module.exports = {
         }
 
         /**
+         * Report location example :
+         *
+         * for unexpected space `before`
+         *
+         * var a = 'b'   ;
+         *            ^^^
+         *
+         * for unexpected space `after`
+         *
+         * var a = 'b';  c = 10;
+         *             ^^
+         *
          * Reports if the given token has invalid spacing.
          * @param {Token} token The semicolon token to check.
          * @param {ASTNode} node The corresponding node of the token.
          * @returns {void}
          */
         function checkSemicolonSpacing(token, node) {
             if (astUtils.isSemicolonToken(token)) {
-                const location = token.loc.start;
 
+                // console.log(token.loc)
                 if (hasLeadingSpace(token)) {
+                    const tokenBefore = sourceCode.getTokenBefore(token);
+                    const loc = {
+                        start: tokenBefore.loc.end,
+                        end: {
+                            line: token.loc.start.line,
+                            column: token.loc.start.column - 1
+                        }
```

Comment:
The end loc is exclusive (it isn't supposed to be underlined), so this would produce:

![image](https://user-images.githubusercontent.com/44349756/82111322-d11b0280-9744-11ea-920e-e31d46eea3b9.png)


It can be just: `end: token.loc.start`.

(similar applies to the `after: false` branch)

Reply:
ahh, I see. it does make things better when being tested in an editor. 

_Just a small question on a different context._

I tried using the `master` branch with `vscode-eslint` using the `file` type dep for `eslint` i.e

`"eslint": file:../<path-to-my-local-eslint>` in this line 

https://github.com/microsoft/vscode-eslint/blob/master/package.json#L451

and opening the vscode's extension debugger doesn't seems to work out for me. did you have any different approach on this?

Reply:
> The end loc is exclusive (it isn't supposed to be underlined),

so when we give both start and end value it does puts a line over that right ?
and if the starting and ending column is same i.e

with this

```js
  end: {
                            line: token.loc.start.line,
                            column: token.loc.start.column - 1
                        }
```
and there is only single space

`var a = 'a' ;`
so it will just underline that space only right ? or it won't underline that?

cause if I make it to 

```js
  end:  token.loc.start
`

for `var a = 'a' ;` it will put a line below the space and for `;` as well right . 

_I am not able to work this our with `vscode-eslint` that's why just need to confirm these_

Reply:
> I tried using the `master` branch with `vscode-eslint` using the `file` type dep for `eslint` i.e
> 
> `"eslint": file:../<path-to-my-local-eslint>` in this line
> 
> https://github.com/microsoft/vscode-eslint/blob/master/package.json#L451
> 
> and opening the vscode's extension debugger doesn't seems to work out for me. did you have any different approach on this?

That line is in `devDependencies`, so I think it's for linting `vscode-eslint` project itself?

Not sure about debugging, but in order to just see how the change looks in the editor you can open a test js project and add `eslint` dependency pointing to local working copy (e.g. `"eslint": "file:../eslint"`, then do `npm install`, also vscode restart is required after each change). Or, install `eslint` normally and overwrite the rule in `node_modules/eslint/lib/rules` (again, vscode restart is required after each change).

Reply:
> That line is in `devDependencies`, so I think it's for linting `vscode-eslint` project itself?

yeah, cause I was testing them on `vscode-eslint` so I thought if I can use the dev dep's eslint to link it with `master` and use it in `vscode-eslint`'s code itself. 



> Not sure about debugging, but in order to just see how the change looks in the editor you can open a test js project and add `eslint` dependency pointing to local working copy (e.g. `"eslint": "file:../eslint"`, then do `npm install`, also vscode restart is required after each change). Or, install `eslint` normally and overwrite the rule in `node_modules/eslint/lib/rules` (again, vscode restart is required after each change).

this seems much better,  thanks I will check this. 👍

Reply:
> cause if I make it to
> 
> ```js
>   end:  token.loc.start
> ```
> 
> for `var a = 'a' ;` it will put a line below the space and for `;` as well right .

It will underline only the space, `end` doesn't count as being a part of the range.

Reply:
ohh, 
so how can I underline the whole space cause it does need the range right

```js
var a = 'a'      ;
           ^^^^^^
```

Reply:
I think something like this would do the work:

```js
{
    start: tokenBefore.loc.end,
    end: token.loc.start
}
```

(the last character doesn't count)

Reply:
done 👍